src/ln/channelmanager.rs

```diff
@@ -1085,12 +1085,26 @@ impl ChainListener for ChannelManager {
 		}
 	}
 
+	/// We force-close the channel without letting our counterparty participate in the shutdown
 	fn block_disconnected(&self, header: &BlockHeader) {
+		let mut closed_channel = Vec::new();
```

Comment:
I think you can skip this, too, if you switch back to the borrow_parts() (since borrow_parts() is used to let you get around the borrow checker for stuff like this :))

Reply:
Sorry are you sure it works with closure? I get a E500 or E0499 each time, I'm still a beginner in magic tricks against borrowck

Reply:
This diff works for me (though no guarantees about older rustcs, lets see what travis says): https://0bin.net/paste/at3mwfdpXIeugIQ4#X62nrP-X/OYnkLEJ75jrbpR9FRY8DxcOudBzraNDW98